crypto/dilithium/p_dilithium3.c

```diff
@@ -5,13 +5,16 @@
 #include <openssl/err.h>
 #include <openssl/mem.h>
 
+#include "../crypto/internal.h"
 #include "../fipsmodule/evp/internal.h"
 #include "../evp_extra/internal.h"
 #include "sig_dilithium.h"
 
 static int pkey_dilithium3_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   DILITHIUM3_KEY *key = OPENSSL_malloc(sizeof(DILITHIUM3_KEY));
-  if (key == NULL) {
+  key->pub = OPENSSL_malloc(DILITHIUM3_PUBLIC_KEY_BYTES);
+  key->priv = OPENSSL_malloc(DILITHIUM3_PRIVATE_KEY_BYTES);
+  if (key == NULL || key->pub == NULL || key->priv == NULL) {
```

Comment:
NP: in the unlikely case the `key` allocation fails you already dereferenced it up on line 15 before this `if key == NULL`.

Reply:
ah, thank you. will fix this and others.